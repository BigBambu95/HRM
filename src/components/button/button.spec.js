import React from 'react'
import sinon from 'sinon'
import Button from './button'

describe('Button', () => {
	let button
	beforeAll(() => {
		button = mount(<Button>Click</Button>)
	})

	it('Рендер кнопки c дефолтными значениями', () => {
		expect(button.text()).toMatch('Click')
	})

	it('Кнопка большого размера', () => {
		button.setProps({ size: 'large' })
		expect(button.props().size).toEqual('large')
		expect(button.find('button').hasClass('large')).toEqual(true)
	})

	it('Кнопка с большим размером шрифта', () => {
		button.setProps({ font: 'large' })
		expect(button.props().font).toEqual('large')
		expect(button.find('button').hasClass('largeFont')).toEqual(true)
	})

	it('Кнопка c кастомным цветом', () => {
		button.setProps({ color: 'red' })
		expect(button.props().color).toEqual('red')
		expect(button.find('button').hasClass('red')).toEqual(true)
	})

	it('Неактивная кнопка', () => {
		button.setProps({ disabled: true })
		expect(button.props().disabled).toEqual(true)
	})

	it('Кнопка с кастомным типом', () => {
		button.setProps({ type: 'submit' })
		expect(button.props().type).toEqual('submit')
	})

	it('Кнопка-иконка', () => {
		button.setProps({ variant: 'icon' })
		expect(button.props().variant).toEqual('icon')
		expect(button.find('button').hasClass('icon')).toEqual(true)
	})

	it('Симуляция клика на кнопку', () => {
		const onButtonClick = sinon.spy()
		const button = mount(<Button onClick={onButtonClick}>Click</Button>)
		button.simulate('click')
		button.simulate('click')
		expect(onButtonClick).toHaveProperty('callCount', 2)
	})
})
