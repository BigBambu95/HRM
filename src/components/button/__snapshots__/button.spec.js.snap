// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Кнопка с переданными пропсами 1`] = `
<button
  className="btn large solid red largeFont fullWidth"
  onClick={[Function]}
  type="submit"
>
  Click
</button>
`;

exports[`Button Рендер кнопки c дефолтными значениями 1`] = `
<button
  className="btn medium outlined purple"
  onClick={[Function]}
  type="button"
/>
`;
